File: app/src/main/java/in/kashewdevelopers/todo/adapters/GroupViewHolder.java
package in.kashewdevelopers.todo.adapters;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import in.kashewdevelopers.todo.databinding.GroupListItemLayoutBinding;

public class GroupViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {

    public View view;
    public GroupListItemLayoutBinding binding;
    public String groupName;

    public GroupViewHolder(View view) {
        super(view);
        this.view = view;
        binding = GroupListItemLayoutBinding.bind(view);

        binding.root.setOnClickListener(this);
        binding.root.setOnLongClickListener(this);
    }

    @Override
    public void onClick(View view) {
        onClick(false);
    }

    @Override
    public boolean onLongClick(View v) {
        onClick(true);
        return true;
    }


    public void onClick(boolean showDetails) {
        if (onItemClickListener != null) {
            onItemClickListener.onItemClickListener(this, showDetails);
        }
    }


    // callbacks
    public interface OnItemClickListener {
        void onItemClickListener(GroupViewHolder holder, boolean showDetail);
    }

    private OnItemClickListener onItemClickListener;

    public GroupViewHolder setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
        return this;
    }

}

File: app/src/main/java/in/kashewdevelopers/todo/adapters/GroupViewHolder.kt
package `in`.kashewdevelopers.todo.adapters

import `in`.kashewdevelopers.todo.databinding.GroupListItemLayoutBinding
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class GroupViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener, View.OnLongClickListener {

    val binding = GroupListItemLayoutBinding.bind(view)
    var groupName: String? = null

    init {
        binding.root.setOnClickListener(this)
        binding.root.setOnLongClickListener(this)
    }

    override fun onClick(view: View) {
        onClick(false)
    }

    override fun onLongClick(view: View): Boolean {
        onClick(true)
        return true
    }


    private fun onClick(showDetails: Boolean) {
        onItemClickListener?.onItemClickListener(this, showDetails)
    }


    // callbacks
    interface OnItemClickListener {
        fun onItemClickListener(holder: GroupViewHolder, showDetail: Boolean)
    }

    private var onItemClickListener: OnItemClickListener? = null

    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?): GroupViewHolder {
        this.onItemClickListener = onItemClickListener
        return this
    }

}
